packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +15,75 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	public caseInsensitiveSubCommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+
+		if (clientOptions.caseInsensitiveCommands) {
+			this.caseInsensitiveSubCommands = true;
+
+			// Because slash commands must be lowercase anyway, we can transform all to lowercase.
+			for (const cmd of this.parsedSubcommandMappings) {
+				cmd.name = cmd.name.toLowerCase();
+				if (cmd.type === 'group') {
+					for (const groupCommand of cmd.entries) {
+						groupCommand.name = groupCommand.name.toLowerCase();
+					}
+				}
+			}
+		}
+
+		if (options.generateDashLessAliases) {
+			for (const mapping of this.parsedSubcommandMappings) {
+				if (!Reflect.has(mapping, 'messageRun')) {
+					continue;
+				}
+
+				const dashLessMappings: SubcommandMappingArray = [];
+
+				if (mapping.type === 'group') {
+					let hasChangedValues = false;
+					const clonedMappingEntries = [...mapping.entries];
+
+					for (const groupCommand of mapping.entries) {
+						if (groupCommand.name.includes('-')) {
+							hasChangedValues = true;
+							clonedMappingEntries.push({
+								...groupCommand,
+								name: groupCommand.name.replaceAll('-', '')
+							});
+						}
+					}
+
+					if (hasChangedValues) {
+						mapping.entries = clonedMappingEntries;
+					}
+
+					if (mapping.name.includes('-')) {
+						hasChangedValues = true;
+						mapping.name = mapping.name.replaceAll('-', '');
```

Comment:
Doesn't this overwrite in place instead of overwriting a clone? 😅 
In turn that means there will never be a group mapping with dashes

Reply:
err... crap.. yeah I think you're right. We need to create a new group instead of replacing it. Crap.